Counting points

shaulfrancus opened the pull request:
Implemented Square multipliers and using them to count points for words. Also rewrote some methods of Board to be less repetitive.

Commit:
added multipliers/modifiers for Squares

Commit:
added isEmpty method for Squares

Commit:
implemented counting points on insert

Commit:
made lettersNeeded and checkWord methods of Board less repetitive

Commit:
Added getting bonus points for using every tile on rack.

Commit:
Made board use squares with modifiers and Board.toString to show which squares hae which modifiers

Commit:
Bug fix

Commit:
Bug fix take 2

Commit:
Bug fix take 3

Commit:
Tiny bug fix

Commit:
Bug fix

yash0024 reviewed (approved):
![image](https://user-images.githubusercontent.com/87946364/141245069-41370fbf-40fc-42e5-96d0-5d78308fe23b.png)
error in line 118 of Board when inserting 'torn' starting at A6, moving down

yash0024 committed:
Several changes

Modfies Board's checkWord method, and added methods containsOnlyOneTile, checkIfConnectedWordIsValid, and letterTouchesAnotherWord to implement the following Scrabble rules:

New words may be formed by:
Adding one or more letters to a word or letters already on the board.

Placing a word at right angles to a word already on the board. The new word must use one of the letters already on the board or must add a letter to it.

Placing a complete word parallel to a word already played so that adjacent letters also form complete words.

Note: The program works if and only if the first move starts at the centre square. That issue must be fixed for the program to work at all times.

yash0024 committed:
fixed bug in Game

code didn't deal with inputs like H14, where the column was >= 10

yash0024 committed:
Refactoring

Refactored stuff

yash0024 committed:
Bug Fix

Bug fix

yash0024 committed:
bug fix

Commit:
Bug fix

Commit:
rewrote containsOnlyOneTile

yash0024 committed:
fixed bug
